### ISSUE TITLE
Handling of np.inf values

### ISSUE THREAD

**Issue by Mosula7** [catboost/catboost#2696]
Problem:
I want to know how how catboost handles np.inf values during training? Are they ignored completely or are they treated as a separate value/category?

Can training a model on data that contains np.inf values cause any problem and is it generally recommended to handle them before training?

I expected that np.inf values would be treated the same as np.nan values, but when I replace np.inf with np.nan in my test set the pretrained model (trained on values with np.inf) is giving different probability outputs.

I couldn't find any information regarding this topic in the documentation.

catboost version: 1.2.3
Operating System: Windows 10
CPU: Intel(R) Core(TM) i5-10500 CPU
GPU:

**Comment by andrey-khropov** [comment]
Nan values processing is determined by `nan_mode` training parameter. See [here](https://catboost.ai/en/docs/concepts/algorithm-missing-values-processing) for details.

The default `nan_mode` is `Min`, it means that nan values are treated as values that are less than all other values of this feature. 

`np.inf` will be considered as larger than other values, CatBoost does not process `np.inf` values in any specific way.

If you select `nan_mode` as `Max` then the result could be similar to the case when `np.nan` is replaced with `np.inf` in the input data, still with `np.nan` or other missing values it is guaranteed that missing values will have a separate discrete value after quantization which is not guaranteed for `np.inf`.
